### PR TITLE
freshness asset check

### PR DESCRIPTION
**Pull request by johannkm**
Sample of what this could look like.

With this I discovered a bug that apparently if you try to observe an asset that has a check, we'll only run the check? Worrisome

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19525** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19525?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19525?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/data_version.py [hunk 1]
```diff
@@ -43,7 +43,10 @@ class UnknownValue:
 class DataVersion(
     NamedTuple(
         "_DataVersion",
-        [("value", str)],
+        [
+            ("value", str),
+            ("data_time", Optional[float]),
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/data_version.py:48]: I think we'll want to put the data time on whatever encloses the `DataVersion`, e.g. `ObserveResult`.  Users who aren't dealing with the versioning system shouldn't need to deal with the data version concept.

Comment (johannkm) [reply python_modules/dagster/dagster/_core/definitions/data_version.py:48]: Agreed. Should you be able to emit `AssetObservation`s from `@observable_source_asset`s? API is a bit odd with the system tags but we could change the  \_\_new\_\_ params

Comment (sryza) [reply python_modules/dagster/dagster/_core/definitions/data_version.py:48]: I don't think so, for the same reason we don't expect users to emit `AssetMaterialization`s from `@asset`s. We use `MaterializeResult` instead, so the analog would be `ObserveResult`.

**Pull request closed without merging**